src/PooledArrays.jl

```diff
@@ -439,40 +446,49 @@ Base.convert(::Type{Array}, pa::PooledArray{T, R, N}) where {T, R, N} = convert(
 ##
 ##############################################################################
 
-# Scalar case
-Base.@propagate_inbounds function Base.getindex(pav::PooledArrOrSub, I::Integer...)
-    idx = DataAPI.refarray(pav)[I...]
-    iszero(idx) && throw(UndefRefError())
-    return @inbounds DataAPI.refpool(pav)[idx]
+# We need separate functions due to dispatch ambiguities
+
+for T in (PooledArray, SubArray{<:Any, <:Any, <:PooledArray})
+    @eval Base.@propagate_inbounds function Base.getindex(A::$T, I::Integer...)
```

Comment:
I can never remember, but doesn't this work without `@eval`?
```suggestion
    Base.@propagate_inbounds function Base.getindex(A::T, I::Integer...)
```

Reply:
I get the following error:
```
julia> for T in [Int, String]
       f(::T) = nothing
       end
ERROR: UndefVarError: T not defined
Stacktrace:
 [1] top-level scope
```

Reply:
I think you need the `@eval` because the for-loop introduces a local scope, so you need to eval your way out